### PR TITLE
Update PR template checklist

### PR DESCRIPTION
**Pull request by jordy25519**
Updated to give relevance to this project

### CODE REVIEW

File: PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,23 +1,20 @@
...
+  - What important points should reviewers know?
   - Is there something left for follow-up PRs?
-- [ ] You labeled the PR with appropriate labels if you have permissions to do so.
-- [ ] You mentioned a related issue if this PR related to it, e.g. `Fixes #228` or `Related #1337`.
-- [ ] You asked any particular reviewers to review. If you aren't sure, start with GH suggestions.
-- [ ] Your PR adheres [the style guide](https://github.com/paritytech/polkadot/wiki/Style-Guide)
+  - For large diffs please include a changelog in the description with a bulleted "Adds | Changes | Fixes | Removes" sections
+- [ ] Add Apache 2.0 license header for new files
+- [ ] Mention the related issue, if any, e.g. `Fixes #228` or `Related #1337`.
+- [ ] Include integration tests, if appropriate
```
Comment (cowboy-bebug) [inline PULL_REQUEST_TEMPLATE.md]: Maybe we could add "Must have unit tests"

File: PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,23 +1,20 @@
...
+- [ ] Request reviewers. If you aren't sure, start with GH suggestions.
+- [ ] Update rustdoc comments with changes
+- [ ] Update README with changes, if appropriate
+- [ ] Adhere to [the style guide](https://github.com/paritytech/polkadot/wiki/Style-Guide)
```
Comment (cowboy-bebug) [inline PULL_REQUEST_TEMPLATE.md]: Apparently the link is moved to here https://wiki.parity.io/Substrate-Style-Guide

File: PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,23 +1,20 @@
...
   - In particular, mind the maximal line length.
+  - Tabbed Indentation
```
Comment (cowboy-bebug) [inline PULL_REQUEST_TEMPLATE.md]: Can we say "Hard Tab Indentation" for further clarity maybe

**Review by cowboy-bebug — CHANGES_REQUESTED** [review]
Nice! I got a few nitpicks

File: PULL_REQUEST_TEMPLATE.md (hunk 3 above)
Comment (jordy25519) [reply PULL_REQUEST_TEMPLATE.md]: I'll remove it as it's mentioned in the style guide

**Review by cowboy-bebug — APPROVED** [review]
Thanks!

**Pull request merged**